PhysicsTools/PatAlgos/plugins/JetCorrFactorsProducer.cc

```diff
@@ -232,12 +232,9 @@ JetCorrFactorsProducer::produce(edm::Event& event, const edm::EventSetup& setup)
 					    << "the sequence.";
     }
     for(unsigned int idx=0; idx<corrLevel->second.size(); ++idx){
-      bool flavorDependent=false;
       std::vector<float> factors;
-      if(flavorDependent ||
-	 corrLevel->second[idx].find("L5Flavor")!=std::string::npos ||
+      if(corrLevel->second[idx].find("L5Flavor")!=std::string::npos ||
```

Comment:
Looking at this code more carefully, I think what we want to do is move the line
```
bool flavorDependent=false;
```
immediately before the `for` statement and leave all the other lines. I'm pretty sure the comment is correct when it says that after first finding L5Flavor/L7Parton, all subsequent correction levels should be flavor-dependent. However, it does rely on the correction level lists being set in order like this:
'L1FastJet', 'L2Relative', 'L3Absolute','L5Flavor', 'L7Parton'
If the input levels to this producer are in the wrong order, then this code would misbehave with my suggested change. It might be safer to go with Usama's version, and the performance improvement when using `flavorDependent` seems trivial.
@slava77, @monttj: Do you have any suggestions?